src/blockrelay/compactblock.cpp

```diff
@@ -1101,21 +1102,27 @@ void SendCompactBlock(ConstCBlockRef pblock, CNode *pfrom, const CInv &inv)
     }
 }
 
-bool IsCompactBlockValid(CNode *pfrom, const CompactBlock &compactBlock)
+bool IsCompactBlockValid(CNode *pfrom, std::shared_ptr<CompactBlock> compactBlock)
 {
     validateCompactBlock(compactBlock);
 
+    // Check that we havn't exceeded the max allowable block size that would be reconstructed from this
+    // set of hashes
+    uint64_t nTxnsInBlock = compactBlock->shorttxids.size() + compactBlock->prefilledtxn.size();
+    if (nTxnsInBlock > (thinrelay.GetMaxAllowedBlockSize() / MIN_TX_SIZE))
+        return error("Number of hashes in thinblock or xthinblock would reconstruct a block the block size limit\n");
```

Comment:
would reconstruct a block **greater than** the block size limit

Reply:
done